component/core/src/main/java/org/exoplatform/social/core/listeners/SocialMembershipListenerImpl.java

```diff
@@ -65,10 +65,9 @@ public void postDelete(Membership m) throws Exception {
         if(state != null && state.getIdentity() != null && space.getEditor() == null) {
           space.setEditor(state.getIdentity().getUserId());
         }
-        boolean hasAllMembership = SpaceUtils.isUserHasMembershipTypesInGroup(m.getUserName(), m.getGroupId(), MembershipTypeHandler.ANY_MEMBERSHIP_TYPE);
-        boolean hasManagerMembership = hasAllMembership || SpaceUtils.isUserHasMembershipTypesInGroup(m.getUserName(), m.getGroupId(), acl.getAdminMSType());
-        boolean hasMemberMembership = hasManagerMembership || SpaceUtils.isUserHasMembershipTypesInGroup(m.getUserName(), m.getGroupId(), SpaceUtils.MEMBER);
-
+        boolean hasAllMembership = orgService.getMembershipHandler().findMembershipByUserGroupAndType(m.getUserName(), m.getGroupId(), MembershipTypeHandler.ANY_MEMBERSHIP_TYPE) != null;
+        boolean hasManagerMembership = hasAllMembership || orgService.getMembershipHandler().findMembershipByUserGroupAndType(m.getUserName(), m.getGroupId(), acl.getAdminMSType()) != null;
+        boolean hasMemberMembership = hasManagerMembership || orgService.getMembershipHandler().findMembershipByUserGroupAndType(m.getUserName(), m.getGroupId(), SpaceUtils.MEMBER) != null;
```

Comment:
This will cause a **big performance degradation**, please consider adding a dedicated listener when a , instead to update user identity inside current ConversationState using IdendityRegistry, which is already made by the listener

https://github.com/Meeds-io/gatein-portal/blob/d8e2afb2d40f5456d19f5e5528dd311a664cadfc/web/portal/src/main/webapp/WEB-INF/conf/organization/organization-configuration.xml#L348-L352

The fix will be to ensure that this Listener do the Job of updating ACL Identity of user after deleting him from space.